rero_ils/modules/patrons/api.py

```diff
@@ -775,6 +779,52 @@ def get_current_patron(record):
             if ptrn.organisation_pid == record.organisation_pid:
                 return ptrn
 
+    @classmethod
+    def set_communication_channel(
+            cls, user=None, dbcommit=True, reindex=True):
+        """Set communication_channel for patrons according to patrons data.
+
+        :param user - user record
+        :param dbcommit - commit the changes
+        :param reindex - index the changes
+        """
+        def basic_query(channel):
+            """Returns basic ES query."""
+            return PatronsSearch()\
+                .filter('term', user_id=user.id)\
+                .filter('term', patron__communication_channel=channel)\
+                .source(includes='pid')
+
+        mail_query = basic_query(CommunicationChannel.EMAIL)\
+            .filter('bool', must_not=[
+                Q('exists', field='patron.additional_communication_email'),
+                Q('exists', field='email')
+            ])
+        to_mail_pids = [[hit['pid'], CommunicationChannel.MAIL, hit.meta.id]
```

Comment:
COSMETIC : Should be better, in a pythonic way to return a 'tuple' instead of 'list'.
```python
to_mail_pids = [(hit['pid'], CommunicationChannel.MAIL, hit.meta.id) for hit in mail_query.scan()]
```